spec/models/program_spec.rb

```diff
@@ -199,4 +200,61 @@
     end
   end
 
+  describe '#any_event_for_this_date?' do
+    context 'without selected_schedule' do
+      it 'returns false if there are events on the date' do
+        event = create(:event_schedule, start_time: Date.today.to_time)
+        program.selected_schedule.event_schedules = [event]
+        expect(program.any_event_for_this_date? '2017-01-01').to eq false
+      end
+
+      it 'returns false if there are no events on the date' do
+        expect(program.any_event_for_this_date? "2017-01-01").to eq false
+      end
+    end
+
+    context 'with selected schedule' do
+      before :each do
+        program.selected_schedule = schedule
+        program.save!
+      end
+
+      it 'returns boolean' do
+        expect(program.any_event_for_this_date?('').in? [true, false] ).to eq true
+      end
+
+      it 'returns false if there are not events today and date is nil' do
+        expect(program.any_event_for_this_date? nil).to eq false
+      end
+
+      it 'returns true if there are events today and date is nil' do
+        event = create(:event_schedule)
+        program.selected_schedule.event_schedules = [event]
+        program.save!
+        expect(program.any_event_for_this_date? nil).to eq true
+      end
+
+      it 'returns false if there are not events today and date is empty' do
```

Comment:
it doesn't make any difference if there are events or not when the date is empty/nil. I would say that it is enough to test it when date is empty and there is an event for that time which is in the selected schedule. Although it is enough with having in the title 'returns false if date is empty'